_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+5-arg `mul!` takes care of all that for you dispatching to `BLAS.gemm!` or other suitable methods once it has that all sorted.
+Further, the existing 3-arg `mul!(C, A, B)` is a special case of it:
+`mul!(C, A, B) = mul!(C, A, B, false, true)` (`true` and `false` being 1, and strong 0).
+So you can just implement the 5-arg form and be done with it.
+
+I personally didn't care about 5-arg `mul!` at all for a long time.
+It was yet another in-place function in `LinearAlgebra` that I would never use often enough to remember what it did, and thus wouldn't use.
+But I realized that it is a crucial function for my own area: automatic differentiation.
+`mul(C, A, B, true, true)` is bthe in-place accumulation rule for the reverse mode equivalent of the product rule.
+
+
+### You can now print and interpolate `nothing` into strings.
+
+This is one of mine [#32148](https://github.com/JuliaLang/julia/pull/32148), and I find it is such a usability enhancement.
+So many small frustrations in Julia 1.0 related to interpolating a variable containing `nothing` into a string.
+Often occurring when you are adding a quick `println` to debug something not being the value you expected; or when building a string for some other error message.
+Arguably both of those are better done via other means (`@show`, and values stored in fields in the error type); but we don't always do what is best.
+Sometimes it is expedient to just interpolate things into strings without worrying about what type they are.
+
+### `Base.download` now using libcurl
+For a very long time the `download` function which retrieves things over HTTP, was implemented with an amazing hack.
+It conditionally shelled out to different programs.
+On Windows it ran a mildly scary PowerShell script.
+On Unixen it first tried to use `curl`, then if that wasn't installed it tried to use `wget`, and then if that wasn't installed it tried to use `fetch`.
+It's low-key amazing that this worked as well as it did -- very few complaints.
+But as of 1.6, [it now uses libcurl](https://github.com/JuliaLang/julia/pull/37340).
+Using libcurl everywhere gives consistency with proxy settings, and protocol support (beyond HTTP) across all platforms.
+
+It also has a more extensive API via the new [Downloads.jl](https://github.com/JuliaLang/Downloads.jl) standard library.
+It can do things like progress logging, and it can retrieve headers.
+I have [tried getting headers via conditional different command-line download functions](https://github.com/oxinabox/DataDeps.jl/pull/22) before, it's a small cup of nightmare fuel; and I ended up swapping out to HTTP.jl for that.
+It wouldn't be too surprising if eventually we see libcurl swapped out for code extracted from HTTP.jl for a pure Julia solution.
```

Comment:
btw, there was some discussion in #web on Slack about going the other way, swapping out the HTTP.jl client code for libcurl, to reduce the maintenance burden

Reply:
Interesting.

Reply:
just checked and the messages still exist (just barely out of the slack hole!), in case didn't see: https://julialang.slack.com/archives/C683N3423/p1612489433044100